cmd/chain-maind/app/app.go

```diff
@@ -142,26 +143,37 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		config.SetRoot(clientCtx.HomeDir)
 		path := config.GenesisFile()
 
-		file, err := os.OpenFile(path, os.O_RDWR, 0644)
+		file, err := os.OpenFile(path, os.O_RDWR, 0600)
+		if !chaingenutilcli.IsValidPath(path) {
+			return fmt.Errorf("insecure filepath %s", path)
+		}
```

Comment:
For this file path finding, two things:
1. it's likely a false positive, as the scope of this is very limited -- at least in the case of genutil
2. this "fix" is a bit problematic:
- it's done after opening the file
- it does some custom input validation with a deny list

So I suggest keeping this simple and either:
- suppress this warning (`// nolint: gosec`)
- or just do a simple path cleanup / canonical path -- `filepath.Clean(path)`